src/ast.ts

```diff
@@ -163,16 +164,16 @@ class TreeAdapter implements vscode.TreeDataProvider<ASTNode> {
     return element.children || [];
   }
 
-  public getParent(node: ASTNode): ASTNode {
+  public getParent(node: ASTNode): ASTNode|undefined {
```

Comment:
There are a bunch of places in this class where you've changed `null` to `undefined` - is there some fundamental reason why null is incorrect? (It makes sense to be consistent within a module of course.

My understanding is this is mostly a stylistic choice, and my preference would be to use null for "deliberately" null values, but this isn't my primary language...

Reply:
No worries at all. I think if `undefined` is privileged by the `?.` operators etc then ergonomics is a good reason to prefer it.
(And it seems so, playground says `(null as any)?.foo` is `undefined`)
And this seems to be MS's preference, and it's mostly their APIs we're dealing with.

Reply:
I agree. So just to confirm, are deciding to use `undefined` for values stored by the extension and accepting `null` for incoming values (when necessary)?

Reply:
That sounds good to me!